### PR TITLE
[Silabs] Add SIT ICD configurations to silabs lock example

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
Enables SIT ICD device types on the silabs door lock example app

* Rename SED to ICD in documentation and build scripts
* Remove Subscription application callback
* Add ICDM cluster on door lock common
* Enable ICD behavior by default on door lock example

### CODE REVIEW

File: third_party/silabs/efr32_sdk.gni [hunk 1]
```diff
@@ -46,10 +46,16 @@ declare_args() {
...
+  # ICD Openthread Configuration flags
   sl_ot_idle_interval_ms = 30000  # 30s Idle Intervals
-  sl_ot_active_interval_ms = 200  # 500ms Active Intervals
-  sl_active_mode_threshold = 1000  # 1s Active mode threshold
+  sl_ot_active_interval_ms = 200  # 200ms Active Intervals
+
+  # ICD Matter Configuration flags
+  sl_idle_mode_interval_ms = 60000  # 10min Idle Mode Interval
+  sl_active_mode_interval_ms = 1000  # 1s Active Mode Interval
+  sl_active_mode_threshold_ms = 500  # 1s Active Mode Threshold
```
Comment (mykrupp) [inline third_party/silabs/efr32_sdk.gni:56]: Fix comment

File: examples/light-switch-app/silabs/README.md [hunk 2]
```diff
@@ -131,9 +131,9 @@ Silicon Labs platform.
 
           ./scripts/examples/gn_silabs_example.sh examples/light-switch-app/silabs/efr32/ out/light-switch-app BRD4187C chip_build_libshell=true
 
-*   Build the example as Sleepy End Device (SED)
+*   Build the example as Intermittently Connected Device (ICD)
 
-          $ ./scripts/examples/gn_silabs_example.sh ./examples/light-switch-app/silabs/efr32/ ./out/light-switch-app_SED BRD4187C --sed
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply examples/light-switch-app/silabs/README.md:136]: done

File: third_party/silabs/efr32_sdk.gni (hunk 1 above)
Comment (mkardous-silabs) [reply third_party/silabs/efr32_sdk.gni]: done

File: third_party/silabs/efr32_sdk.gni [hunk 3]
```diff
@@ -46,10 +46,16 @@ declare_args() {
...
 
-  # ICD Configuration flags
+  # ICD Openthread Configuration flags
   sl_ot_idle_interval_ms = 30000  # 30s Idle Intervals
-  sl_ot_active_interval_ms = 200  # 500ms Active Intervals
-  sl_active_mode_threshold = 1000  # 1s Active mode threshold
+  sl_ot_active_interval_ms = 200  # 200ms Active Intervals
+
+  # ICD Matter Configuration flags
+  sl_idle_mode_interval_ms = 60000  # 10min Idle Mode Interval
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply third_party/silabs/efr32_sdk.gni]: done

**Review by lpbeliveau — APPROVED** [review]

**Review by rcasallas-silabs — APPROVED** [review]

File: examples/lock-app/lock-common/lock-app.matter [hunk 4]
```diff
@@ -1662,6 +1662,30 @@ server cluster UserLabel = 65 {
   readonly attribute int16u clusterRevision = 65533;
 }
 
+/** Allows servers to ensure that listed clients are notified when a server is available for communication. */
+server cluster IcdManagement = 70 {
```
Comment (bzbarsky-apple) [inline examples/lock-app/lock-common/lock-app.matter:1666]: So this is enabling this across all lock apps, right, including the Linux one we use for cert?

How dies it behave in the Linux lock-app, given lack of any obvious implementation in this PR?

Comment (mkardous-silabs) [reply examples/lock-app/lock-common/lock-app.matter:1666]: it uses the default values available in ChipConfig for the cluster. A part from that, adding the attributes of the cluster has no effect.

File: third_party/silabs/efr32_sdk.gni [hunk 5]
```diff
@@ -46,10 +46,16 @@ declare_args() {
   # Enable Segger System View
   use_system_view = false
 
-  # ICD Configuration flags
-  sl_ot_idle_interval_ms = 30000  # 30s Idle Intervals
-  sl_ot_active_interval_ms = 200  # 500ms Active Intervals
-  sl_active_mode_threshold = 1000  # 1s Active mode threshold
+  # ICD Openthread Configuration flags
```
Comment (mykrupp) [inline third_party/silabs/efr32_sdk.gni:49]: It looks like sl_ot_idle_interval_ms sl_ot_idle_interval_ms aren't part of the ICD cluster data model and spec. Are these values described elsewhere?

**Review by mykrupp — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**